### PR TITLE
feat(ip filter) Add ip filter resource for both monitor and secure

### PR DESCRIPTION
**Pull request by vojindj**

### CODE REVIEW

File: sysdig/internal/client/v2/allowed_ip_ranges.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+func (client *Client) DeleteAllowedIpRange(ctx context.Context, id int) error {
+	response, err := client.requester.Request(ctx, http.MethodDelete, client.GetAllowedIpRangeURL(id), nil)
+	if err != nil {
+		return err
+	}
+	defer response.Body.Close()
+
+	if response.StatusCode != http.StatusNoContent && response.StatusCode != http.StatusOK && response.StatusCode != http.StatusNotFound {
```
Comment (filiptubic) [inline sysdig/internal/client/v2/allowed_ip_ranges.go:101]: Why having `response.StatusCode != http.StatusNoContent && response.StatusCode != http.StatusOK`? I assume server responds only with StatusOK?

File: sysdig/internal/client/v2/allowed_ip_ranges.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+var AllowedIpRangeNotFound = errors.New("IP range not found")
+
+const (
+	AllowedIPRangesPath = "%s/platform/v1/allowed-ip-ranges"
+	AllowedIPRangePath  = "%s/platform/v1/allowed-ip-ranges/%d"
+)
+
+type AllowedIPRangesInterface interface {
+	Base
+	GetAllowedIpRangeById(ctx context.Context, id int) (*AllowedIpRange, error)
```
Comment (filiptubic) [inline sysdig/internal/client/v2/allowed_ip_ranges.go:19]: In go we should name this `GetAllowedIPRangeByID` this comment applies for all other methods

File: sysdig/internal/client/v2/model.go [hunk 3]
```diff
@@ -174,6 +174,14 @@ type GroupMappingConfig struct {
 	DifferentTeamSameRoleStrategy string `json:"differentRolesSameTeamStrategy"`
 }
 
+type AllowedIpRange struct {
+	ID          int    `json:"id,omitempty"`
+	IpRange     string `json:"ipRange"`
```
Comment (filiptubic) [inline sysdig/internal/client/v2/model.go:179]: IPRange

File: sysdig/internal/client/v2/allowed_ip_ranges.go (hunk 1 above)
Comment (vojindj) [reply sysdig/internal/client/v2/allowed_ip_ranges.go]: Actually, StatusNoContent is returned, so I removed check for StatusOK.

File: sysdig/internal/client/v2/allowed_ip_ranges.go (hunk 2 above)
Comment (vojindj) [reply sysdig/internal/client/v2/allowed_ip_ranges.go]: updated

File: sysdig/internal/client/v2/model.go (hunk 3 above)
Comment (vojindj) [reply sysdig/internal/client/v2/model.go]: updated

**Review by dbonf — APPROVED** [review]
LGTM, I get that we are not able to test the write path of `sysdig_ip_filtering_settings` in integration tests otherwise we cut ourself out of the environment, we can live with that.

**Review by Shadow649 — APPROVED** [review]

**Pull request merged**